src/transformers/models/align/processing_align.py

```diff
@@ -41,12 +45,59 @@ class AlignProcessor(ProcessorMixin):
 
     def __init__(self, image_processor, tokenizer):
         super().__init__(image_processor, tokenizer)
-
-    def __call__(self, text=None, images=None, padding="max_length", max_length=64, return_tensors=None, **kwargs):
+        self.processing_kwargs: ProcessingKwargs = {
+            "common_kwargs": {"return_tensors": None},
+            "text_kwargs": {
+                "text_pair": None,
+                "text_target": None,
+                "text_pair_target": None,
+                "add_special_tokens": True,
+                "padding": "max_length",
+                "truncation": True,
+                "max_length": 64,
+                "stride": 0,
+                "is_split_into_words": False,
+                "pad_to_multiple_of": None,
+                "return_token_type_ids": None,
+                "return_attention_mask": None,
+                "return_overflowing_tokens": False,
+                "return_special_tokens_mask": False,
+                "return_offsets_mapping": False,
+                "return_length": False,
+                "verbose": True,
+            },
+            "images_kwargs": {
+                "do_crop_margin": None,
+                "do_resize": None,
+                "size": None,
+                "resample": None,
+                "do_thumbnail": None,
+                "do_align_long_axis": None,
+                "do_pad": None,
+                "do_rescale": None,
+                "rescale_factor": None,
+                "do_normalize": None,
+                "image_mean": None,
+                "image_std": None,
+                "data_format": "channels_first",
+                "input_data_format": None,
+            },
+            "audio_kwargs": {},
+            "videos_kwargs": {},
+        }
```

Comment:
The only thing I'd say about the current design is the default behaviour of a processor isn't configurable. As users who use multimodal models just want to use the processor, and processors might share the default processing classes they bundle together e.g. CLIP, we might want to be able to override this in the init, such that I can save and load processors with custom behaviour. For example: 

```py
from transformers import AlignProcessor

processor = AlignProcessor.from_pretrained("kakaobrain/align-base")
# I have to pass in max_length in for every call
output = processor(images=images, text=text, max_length=128)

processor = AlignProcessor.from_pretrained("kakaobrain/align-base", max_length=128)
processor.save_pretrained("my_new_model")

processor = AlignProcessor.from_pretrained("my_new_model")
# The processor automatically defaults to the max value I want when calling
output = processor(images=images, text=text)
```

The correct way, might be to say that the e.g. tokenizer behaviour should be configured instead. At the moment, I think if I change the tokenizer, and build the processor, it will use the processor's max_length default, rather than the tokenizer. 

```py
from transformers import AlignProcessor, AutoTokenizer, AutoImageProcessor

tokenizer = AutoTokenizer.from_pretrained("kakaobrain/align-base", max_length=128)
image_processor = AutoImageProcessor.from_pretrained("kakaobrain/align-base")
processor = AlignProcessor(tokenizer, image_processor)

# Is max_length 128 or 64 used here? 
outputs = processor(text=text, images=images)
```

Reply:
That's a good point - I tried the second example with `model_max_length` and it seems to work fine:
![image](https://github.com/huggingface/transformers/assets/39954772/21ac9675-4b39-4551-b8d6-68e42829c2b0)